### ISSUE TITLE
`terraform-provider-azurerm`: refactor to use the new base layer service-by-service

### ISSUE THREAD

**Issue by katbyte** [hashicorp/pandora#1123]
Depends on #1118

**Comment by tombuildsstuff** [comment]
Tracked here: https://github.com/hashicorp/pandora/blob/main/tools/generator-go-sdk/internal/cmd/generate.go#L57 - closing in favour of that list
